src/data/multiset/basic.lean

```diff
@@ -586,6 +586,20 @@ by simp only [repeat_succ, singleton_eq_cons, eq_self_iff_true, repeat_zero, con
 
 theorem eq_of_mem_repeat {a b : α} {n} : b ∈ repeat a n → b = a := eq_of_mem_repeat
 
+lemma repeat_left_injective (n : ℕ) (h : n ≠ 0) : function.injective (λ a : α, repeat a n) :=
+begin
+  intros a b x,
+  simp only at x,
+  apply @eq_of_mem_repeat _ b a n,
+  cases n,
+  { exact (h rfl).elim },
```

Comment:
I think there's some kind of `eq_succ_of_ne` lemma that eliminates the cases here

Reply:
```lean
lemma repeat_left_injective (n : ℕ) (h : n ≠ 0) : function.injective (λ a : α, repeat a n) :=
begin
  intros a b x,
  dsimp only at x,
  obtain ⟨m, rfl⟩ := exists_eq_succ_of_ne_zero h,
  apply eq_of_mem_repeat,
  rw [x.symm, repeat_succ],
  apply mem_cons_self,
end
```